pandas/core/frame.py

```diff
@@ -1978,7 +1978,48 @@ def memory_usage(self, index=True, deep=False):
         return result
 
     def transpose(self, *args, **kwargs):
-        """Transpose index and columns"""
+        """
+        Transpose index and columns.
+
+        Reflect the dataframe over its main diagonal by writing rows as columns
+        and vice-versa.
+
+        Parameters
+        ----------
+        *args
+            These arguments will be passed to
+            :func:`pandas.compat.numpy.function.validate_transpose`.
+        **kwargs
+            Keyword arguments used to create the transposed DataFrame.
+
+        Returns
```

Comment:
can you also add an example with a mixed dtypes, and show the dtypes before/after. and add in the Notes section how transposing can change the dtypes.

Reply:
Hi, @jreback . I've made some changes to incorporate your suggestion.